### DISCUSSION TITLE
How does Gauge tell CI/CD the test result rate?

### DISCUSSION THREAD

**Discussion by jkchua** [getgauge/gauge#2272] [General]
I would like to understand if Gauge has the particular feature to report the results to the CI/CD to help decide if deployment to a higher environment will proceed or not or if it is a custom code that we can develop.

For context and some vague recalling, Azure DevOps has this tab on results where it shows the rate of test cases that have passed and enumerates all test cases that have failed in the process in a row format. The test framework is also Gauge when that test rate is shown. I am unsure if it is Gauge that supplies the metrics to the platform, and if it is indeed Gauge, how can I be able to send the results to the CI/CD to, for example, Jenkins? I am imagining that Gauge has an API that enables that.

**Example**: I ran 100 tests and 60 tests only passed. This means I only get a 60% pass rate, then I would like to enumerate all 40 tests that failed. 

**UPDATE:** I saw the XML Report plugin generating a JUnit XML file and found out that it is being used by Azure and Jenkins to publish test results. My question is, how can Gauge tell that plugin the details of all the test results? Is there any API that Gauge uses to produce those values, and also that we can use too?

**Comment by zabil** [comment]
Yes, there is an example project that you use to start building reports
https://github.com/getgauge-contrib/report-seed

This has details of the hooks you can use. Although the example is in java you can use any language to build something similar.

**Reply by jkchua** [reply]
Looks like this is something helpful. Let me check and report back to you tomorrow. =)
